src/server/game/Battlegrounds/Zones/BattlegroundIC.cpp

```diff
@@ -381,21 +382,23 @@ bool BattlegroundIC::UpdatePlayerScore(Player* player, uint32 type, uint32 value
     return true;
 }
 
-void BattlegroundIC::FillInitialWorldStates(WorldPacket& data)
+void BattlegroundIC::FillInitialWorldStates(WorldPackets::WorldState::InitWorldStates& packet)
 {
-    data << uint32(BG_IC_ALLIANCE_RENFORT_SET) << uint32(1);
-    data << uint32(BG_IC_HORDE_RENFORT_SET) << uint32(1);
-    data << uint32(BG_IC_ALLIANCE_RENFORT) << uint32(factionReinforcements[TEAM_ALLIANCE]);
-    data << uint32(BG_IC_HORDE_RENFORT) << uint32(factionReinforcements[TEAM_HORDE]);
+    packet.Worldstates.emplace_back(BG_IC_ALLIANCE_RENFORT_SET, 1);
```

Comment:
Don't you need to reserve at every function?

Reply:
I had to look this up again, I originally wrote reserve if there's like 7 or so emplace_backs in a row. But, I learned that it's better to follow "Always reserve if the size is known". The lower the amount, the lower the gains, but it's still better :+1:  Will have another look to add reserve